zdm/energetics.py

```diff
@@ -10,20 +10,236 @@
 SplineMin = -6
 SplineMax = 6
 NSpline = 1000
+SplineLog = True
 
 ############## this section defines different luminosity functions ##########
 
-def init_igamma_splines(gammas, reinit=False):
-    global SplineMin,SplineMax,NSpline
+def init_igamma_splines(gammas, reinit=False,k=3):
+    """
+    gammas [list of floats]: list of values of gamma at which splines
+        must be created
+    reinit [bool]: if True, will re-initialise even if a spline for
+        that gamma has already been created
+    k [int]: degree of spline to use. 3 by default (cubic splines).
+        Formal range: integers 1 <= k <= 5. Do NOT use 2 or 4.
+    
+    If SplineLog is set, interpolations are performed in log-space,
+        i.e. the results is a spline interpolation of the log10 of the
+        answer in terms of the log10 of the input
+    """
+    global SplineMin,SplineMax,NSpline,SplineLog
     for gamma in gammas:
         if gamma not in igamma_splines.keys() or reinit:
             print(f"Initializing igamma_spline for gamma={gamma}")
-            avals = 10**np.linspace(SplineMin, SplineMax, NSpline)
+            lavals = np.linspace(SplineMin, SplineMax, NSpline)
+            avals = 10**lavals
             numer = np.array([float(mpmath.gammainc(
                 gamma, a=iEE)) for iEE in avals])
-            # iGamma
-            igamma_splines[gamma] = interpolate.splrep(avals, numer,k=3)
+            if SplineLog:
+                # check for literal zeros, set them to small values
+                zero = np.where(numer == 0.)[0]
+                ismall = zero[0]-1
+                smallest = numer[ismall]
+                numer[zero] = smallest
+                lnumer = np.log10(numer)
+                igamma_splines[gamma] = interpolate.splrep(lavals, lnumer,k=k)
+            else:
+                igamma_splines[gamma] = interpolate.splrep(avals, numer,k=k)
+            
+
+
+def test_spline_accuracy(gamma = -1.5837, Ntest = 100):
```

Comment:
please move this to a file in the tests/ folder

Reply:
extra credit if you get it to run under CI

Reply:
Done